sympy/core/tests/test_numbers.py

```diff
@@ -487,14 +488,30 @@ def teq(a):
     assert Float(decimal.Decimal('-Infinity')) == S.NegativeInfinity
 
     assert '{0:.3f}'.format(Float(4.236622)) == '4.237'
-    assert '{0:.35f}'.format(Float(pi.n(40), 40)) == '3.14159265358979323846264338327950288'
+    assert '{0:.35f}'.format(Float(pi.n(40), 40)) == \
+        '3.14159265358979323846264338327950288'
 
     assert Float(oo) == Float('+inf')
     assert Float(-oo) == Float('-inf')
 
     # unicode
-    assert Float(u'0.73908513321516064100000000') == Float('0.73908513321516064100000000')
-    assert Float(u'0.73908513321516064100000000', 28) == Float('0.73908513321516064100000000', 28)
+    assert Float(u'0.73908513321516064100000000') == \
+        Float('0.73908513321516064100000000')
+    assert Float(u'0.73908513321516064100000000', 28) == \
+        Float('0.73908513321516064100000000', 28)
+
+    # binary precision
+    # Decimal value 0.1 cannot be expressed precisely as a base 2 fraction
+    Float(S(1)/10, precision=53) == S(0.1)
+    Float(S(1)/10, precision=54) == S(0.1)
+    assert not Float(S(1)/10, precision=53) == Float(S(1)/10, precision=54)
```

Comment:
I think explicitly testing for this bullshit non-associate structural equality is a bad idea.  i.e., this is `a == b` and `c == b` but `a != c`.  This isn't your fault, but let's not build it into the tests.

Would this be a better test?
````

In [34]: a = Float(S(1)/10, dps=15)

In [35]: b = Float(S(1)/10, dps=16)

In [36]: p = Float(S(1)/10, precision=53)

In [37]: q = Float(S(1)/10, precision=54)

In [38]: a._mpf_ == p._mpf_
Out[38]: True

In [39]: a._mpf_ == q._mpf_
Out[39]: False

In [40]: b._mpf_ == q._mpf_
Out[40]: False
````
This shows that:
1.  `a` and `p` are the same
2.  But there are some bit precisions with no corresponding `dps`.  (`q` is "between" `a` and `b`).